### PR TITLE
feat(issue-stream): Find search from backend 

### LINKED ISSUE #57853: Update issues backend to handle default search
Update the issues backend to handle looking up saved searches and finding the default search, if it exists.

### PR DESCRIPTION
**Pull request by roggenkemper**
this pr allows the main issues request from the issue stream to be made without waiting for the saved searches request. 2 additional parameters are optionally passed to the backend: one if there is a specific saved search and one if saved searches have not loaded from the front end. If saved searched are not loaded from the front end, it will check if there is a specific saved search requested. If there is, it will be used. If not, it will check if the user has a default search set, and use it if it is found. If not, it will used the default "is:unresolved" query. 

Closes https://github.com/getsentry/sentry/issues/57853

### CODE REVIEW

File: src/sentry/api/helpers/group_index/index.py [hunk 1]
```diff
@@ -77,6 +79,34 @@ def build_query_params_from_request(
         except ValueError:
             raise ParseError(detail="Invalid cursor parameter.")
     query = request.GET.get("query", "is:unresolved").strip()
+    if request.GET.get("savedSearch") == "0":
```
Comment (scttcper) [inline src/sentry/api/helpers/group_index/index.py:82]: should we check for request.user here

File: src/sentry/api/helpers/group_index/index.py [hunk 2]
```diff
@@ -77,6 +79,34 @@ def build_query_params_from_request(
...
+            )
+            .filter(
+                Q(organization=organization) | Q(is_global=True),
+            )
+            .extra(order_by=["name"])
+        )
+        selected_search_id = request.GET.get("searchId", None)
+        if selected_search_id:
+            # saved search requested by the id
+            saved_search = saved_searches.filter(id=int(selected_search_id))
```
Comment (scttcper) [inline src/sentry/api/helpers/group_index/index.py:100]: if we use .first higher up we can skip .exists which makes an additional sql query
```suggestion
            saved_search = saved_searches.filter(id=int(selected_search_id)).first()
```

**Comment by scttcper** [comment]
would be good to have a basic test here

File: src/sentry/api/helpers/group_index/index.py [hunk 3]
```diff
@@ -77,6 +79,34 @@ def build_query_params_from_request(
...
+        else:
+            # pinned saved search
+            saved_search = saved_searches.filter(visibility=Visibility.OWNER_PINNED)
+
+        if saved_search.exists():
+            search = saved_search.first()
+            query_kwargs["sort_by"] = search.sort
+            query = search.query
```
Comment (scttcper) [inline src/sentry/api/helpers/group_index/index.py:108]: What does this query for if the saved_search isn't found?

Comment (roggenkemper) [reply src/sentry/api/helpers/group_index/index.py:108]: yup, it'll use the default is:unresolved query

**Review by scttcper — APPROVED** [review]

**Pull request merged**